packages/scripts/src/commands/app-data/postProcess/assets.ts

```diff
@@ -238,66 +255,58 @@ export class AssetsPostProcessor {
    */
   private listAssetOverrides(sourceAssets: IContentsEntryHashmap) {
     const entries: IAssetEntriesByType = {
-      global: {},
-      translations: {},
-      themeVariations: {},
-      missing: {},
+      complete: {},
+      orphaned: {},
     };
     const globalFolder = ASSETS_GLOBAL_FOLDER_NAME;
 
     // split assets to separate global, translated and theme assets
     Object.entries(sourceAssets).forEach(([relativePath, entry]) => {
+      const assetEntry = this.contentsToAssetEntry(entry);
+
       let [baseFolder, ...nestedPaths] = relativePath.split("/");
       let nestedPath = nestedPaths.join("/");
-      const assetEntry = this.contentsToAssetEntry(entry);
-      if (baseFolder === globalFolder) {
-        entries.global[nestedPath] = assetEntry;
-      } else if (baseFolder.startsWith("theme_")) {
-        baseFolder = baseFolder.replace("theme_", "");
-        nestedPath = nestedPath.replace(`${ASSETS_GLOBAL_FOLDER_NAME}/`, "");
-        entries.themeVariations[baseFolder] ??= {};
-        entries.themeVariations[baseFolder][nestedPath] = assetEntry;
+      let themeName = "default";
+
+      // Handle case where asset is nested inside a theme folder
+      if (isThemeAssetsFolderName(baseFolder)) {
+        themeName = getThemeNameFromThemeAssetFolderName(baseFolder);
+        [baseFolder, ...nestedPaths] = nestedPath.split("/");
+        nestedPath = nestedPaths.join("/");
+      }
+
+      if (baseFolder === globalFolder || isCountryLanguageCode(baseFolder)) {
+        entries.complete[nestedPath] ??= {} as any;
+        entries.complete[nestedPath].themeVariations ??= {};
+        entries.complete[nestedPath].themeVariations[themeName] ??= {};
+        entries.complete[nestedPath].themeVariations[themeName][baseFolder] = assetEntry;
```

Comment:
Is there a better way to do this (set a property on a nested object that may not already exist)?

Reply:
Not using vanilla JS, but a while back I created a utility `setNestedProperty` function for something similar. 
I've just updated and added some spec tests in #1812 if you want to check it out and see if it will work for your purpose (feel free to add to the tests if there's anything extra you want to include)